yg/emanate/observers.py

```diff
@@ -112,9 +112,10 @@ def _process_event(self, event):
         properties = Properties()
         if event.context:
             properties.headers = event.context
+        routing_key = properties.headers.pop('routing_key', '')
```

Comment:
I recommend making two tweaks to this line:

```python
routing_key = properties.headers.get('routing_key', event.event_type)
```
so that (1) any routing_key value is preserved in the message headers, and (2) the event_type is used as the default routing_key.